internal/backends/sqlite/metadata/pool/pool.go

```diff
@@ -244,3 +244,34 @@ func (p *Pool) Drop(ctx context.Context, name string) bool {
 
 	return true
 }
+
+// InTransaction wraps the given function f in a SQLite transaction.
+//
+// If f returns an error, the transaction is rolled back.
+//
+// Passed context will be used for transaction.
+// Context cancellation DOES rollback the transaction.
+func InTransaction(ctx context.Context, db *sql.DB, f func(*sql.Tx) error) (err error) {
```

Comment:
So, for no `InTransaction` itself would be enough? We don't need the one with retries yet?

Reply:
We need unit test for this though! Let's at least cover basic scenarios for `f` producing error.

Reply:
Why do we need retries?

Reply:
If a transaction fails because of a concurrent query, we might want to retry it.
I'm asking if we need it because I don't see if we can work with concurrent queries yet.